Extend configuration format for segment_range to allow for multiple r…

sebageek opened the pull request:
…anges

segment_range can now contain multiple segment ids separated by ',',
also allowing for single items to appear.

E.g. segment_range = 23,42:45,77

sebageek committed:
Extend configuration format for segment_range to allow for multiple ranges

segment_range can now contain multiple segment ranges separated by ','

E.g. segment_range = 23:23,42:45

abattye commented:
I would prefer not treat this as a special case and instead extend the driver to allow non-contiguous ranges so instead of `2008,2500:3000` we set `2008:2008,2500:3000` and this would then allow us to add additional ranges in future. 

We also need to ensure that the current implementation is not assuming a single continuous range when added the config into the allocations table and when allocating  VLANs from it. This case will not have been specifically tested

sebageek commented:
I removed the special case and now everything is treated as a range. Regarding the "assuming a continuous range" you were right, there was an oversight of mine where `host_config['segment_range']` was being used. I replaced that now with a call to `_segmentations_id` and a membership test. I have found no additional places where this is used, but do you maybe know of any?

Additionally I added a test case for the modified `segmentation_ids()` function to test that it supports single and multiple ranges.

notandy reviewed (approved):
lgtm, like it. Woul backport it to queens as soon as merged.

sebageek commented:
will create a new PR for queens
